source/scpd/Cpp.d

```diff
@@ -113,7 +115,7 @@ extern(C++, (StdNamespace)) {
         else
             private alias TPtr = T*;
 
-        mixin CPPBindingMixin!(shared_ptr!T);
+        mixin CPPBindingMixin!(typeof(this));
```

Comment:
You could also use:
```suggestion
        mixin CPPBindingMixin!(shared_ptr);
```

As the template name refers to the current instantiation. I'm curious why the change was necessary though ?

Reply:
Ah some of the instantiations only partially passed the parameters (not all of the template parameters). I'll use the template name trick though.